### PR TITLE
[rooch-server] Refactor rooch-server, define rooch executor sequencer

### PR DESCRIPTION
**Pull request by jolestar**
part of #41 

1. Define TransactionInfo
2. Move executor to rooch-executor crate.
3. Define rooch-sequencer crate.

Transaction flow:

```
client --- submit tx --> Rpc Service
                                 ---- validate tx ---> Executor
                                 ---- sequence tx ----> Sequencer
                                 ---- execute tx ---> Executor
client <---- response -----
```

Next:
1. Define Proposer and State commitment Chain. The State commitment Chain is a chain of TransactionInfo.

### CODE REVIEW

File: crates/rooch-sequencer/src/actor/messages.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    pub tx: TypedTransaction,
+}
+
+#[derive(Debug)]
+pub struct TransactionSequenceResult {
+    /// The tx order
+    pub order: u128,
+    /// The tx order witness, usually it is the signature of the sequencer to prove the order
+    //TODO confirm the type.
+    pub order_witness: Authenticator,
```
Comment (templexxx) [inline crates/rooch-sequencer/src/actor/messages.rs:20]: order_witness is for verifying tx & its order is correct in sequencer when someone wants to chanllege sequencer.

For normal tx request is just a signature for telling users that sequencer has confirmed that the tx has this order.

@jolestar @steelgeek091

Comment (jolestar) [reply crates/rooch-sequencer/src/actor/messages.rs:20]: I put the signature here and called `order_witness`, the sequencer witness the tx order. In the challenge scenario, it should be `sequence_proof`, which proves the witness is correct.

Comment (templexxx) [reply crates/rooch-sequencer/src/actor/messages.rs:20]: > I put the signature here and called `order_witness`, the sequencer witness the tx order. In the challenge scenario, it should be `sequence_proof`, which proves the witness is correct.

@jolestar 

membership/non-membership witness is proper noun for checking element existed or not in accumulator. So I think witness is not a good idea name here.

And for sequence_proof: Will it make people confused that this sequence and other sequence(it's widely used in other blockchain interface)?

Comment (jolestar) [reply crates/rooch-sequencer/src/actor/messages.rs]: I change the `order_witness` to `tx_order_signature`.

Comment (jolestar) [reply crates/rooch-sequencer/src/actor/messages.rs]: Upon reading numerous articles, I have observed the mix use of "witness" and "proof" which let me confused.

I ask the ChatGPT, and its answer is:
--------------------------------------------
In the accumulator scenario, the terms "membership witness" and "membership proof" are usually used to describe proving whether an element belongs to a specified accumulator. In this scenario, both terms can represent narrowing down the likelihood of "the element belongs to the accumulator".

Generally, "membership witness" refers to the proof that contains the specified element (as in Merkle tree), while "membership proof" refers to the complete proof path.

Therefore, if you only need to prove whether an element belongs to an accumulator, the term "membership witness" may be more intuitive. If you need the complete proof path, the term "membership proof" is more appropriate.

**Comment by baichuan3** [comment]
The role of the current rooch-executor craet, in addition to providing the actor mode, is similar to the block executor in Sui and Aptos, and finally executes the contract transaction, right? @jolestar

**Comment by jolestar** [comment]
> The role of the current rooch-executor crate, in addition to providing the actor mode, is similar to the block executor in Sui and Aptos, and finally executes the contract transaction, right? @jolestar

yes.

**Review by baichuan3 — APPROVED** [review]

**Pull request merged**